Panic with gorilla websocket

I have a 'bot plugin that sends ~4k replies, which is the limit of what slack allows. With the latest nlopes/slack, the bot panics with the dump shown below. With my local fork reverting the gorilla websocket commit, I get my loooong messages as expected. If I get time this weekend I may try to dig deeper, but figured folks might want a heads-up. My code just uses the RTM.SendMessage(s.conn.NewTypingMessage(chanID)) for sending messages.

Here's the full panic / stack trace.
panic: repeated read on failed websocket connection

goroutine 36 [running]:
github.com/uva-its/gopherbot/vendor/github.com/gorilla/websocket.(*Conn).NextReader(0xc042044f00, 0xc042430358, 0x0, 0x0, 0xc04234fec0, 0xc04234fec0)
        C:/Users/dlp7y/go/src/github.com/uva-its/gopherbot/vendor/github.com/gorilla/websocket/conn.go:958 +0x3e0
github.com/uva-its/gopherbot/vendor/github.com/gorilla/websocket.(*Conn).ReadJSON(0xc042044f00, 0x8b65c0, 0xc0422b15a0, 0xba51c0, 0xbd6d30)
        C:/Users/dlp7y/go/src/github.com/uva-its/gopherbot/vendor/github.com/gorilla/websocket/json.go:45 +0x36
github.com/uva-its/gopherbot/vendor/github.com/nlopes/slack.(*RTM).receiveIncomingEvent(0xc042414640)
        C:/Users/dlp7y/go/src/github.com/uva-its/gopherbot/vendor/github.com/nlopes/slack/websocket_managed_conn.go:288 +0xb1
github.com/uva-its/gopherbot/vendor/github.com/nlopes/slack.(*RTM).handleIncomingEvents(0xc042414640, 0xc04236d260)
        C:/Users/dlp7y/go/src/github.com/uva-its/gopherbot/vendor/github.com/nlopes/slack/websocket_managed_conn.go:223 +0x34
created by github.com/uva-its/gopherbot/vendor/github.com/nlopes/slack.(*RTM).ManageConnection
        C:/Users/dlp7y/go/src/github.com/uva-its/gopherbot/vendor/github.com/nlopes/slack/websocket_managed_conn.go:54 +0x26c

--- a/websocket_managed_conn.go
+++ b/websocket_managed_conn.go
@@ -3,7 +3,6 @@ package slack
 import (
 	"encoding/json"
 	"fmt"
-	"io"
 	"net/http"
 	"reflect"
 	"time"
@@ -286,13 +285,8 @@ func (rtm *RTM) ping() error {
 func (rtm *RTM) receiveIncomingEvent() {
 	event := json.RawMessage{}
 	err := rtm.conn.ReadJSON(&event)
-	if err == io.EOF {
-		// EOF's don't seem to signify a failed connection so instead we ignore
-		// them here and detect a failed connection upon attempting to send a
-		// 'PING' message
-
-		// trigger a 'PING' to detect pontential websocket disconnect
-		rtm.forcePing <- true
+	if websocket.IsCloseError(err, websocket.CloseAbnormalClosure) {
+		rtm.killChannel <- false
 		return
 	} else if err != nil {
 		rtm.IncomingEvents <- RTMEvent{"incoming_error", &IncomingEventError{

--- a/websocket_managed_conn.go
+++ b/websocket_managed_conn.go
@@ -3,6 +3,7 @@ package slack
 import (
 	"encoding/json"
 	"fmt"
+	"io"
 	"net/http"
 	"reflect"
 	"time"
@@ -285,21 +286,27 @@ func (rtm *RTM) ping() error {
 func (rtm *RTM) receiveIncomingEvent() {
 	event := json.RawMessage{}
 	err := rtm.conn.ReadJSON(&event)
-	if websocket.IsCloseError(err, websocket.CloseAbnormalClosure) {
+	switch {
+	case err == io.EOF:
+		// EOF's don't seem to signify a failed connection so instead we ignore
+		// them here and detect a failed connection upon attempting to send a
+		// 'PING' message
+
+		// trigger a 'PING' to detect potential websocket disconnect
+		rtm.forcePing <- true
+	case websocket.IsCloseError(err, websocket.CloseAbnormalClosure):
 		rtm.killChannel <- false
-		return
-	} else if err != nil {
+	case err != nil:
 		rtm.IncomingEvents <- RTMEvent{"incoming_error", &IncomingEventError{
 			ErrorObj: err,
 		}}
 		// force a ping here too?
-		return
-	} else if len(event) == 0 {
+	case len(event) == 0:
 		rtm.Debugln("Received empty event")
-		return
+	default:
+		rtm.Debugln("Incoming Event:", string(event[:]))
+		rtm.rawEvents <- event
 	}
-	rtm.Debugln("Incoming Event:", string(event[:]))
-	rtm.rawEvents <- event
 }
 
 // handleRawEvent takes a raw JSON message received from the slack websocket

